src/isDeepEqual.ts

```diff
@@ -174,3 +178,43 @@ function isDeepEqualMaps(
 
   return true;
 }
+
+function isDeepEqualSets(
+  data: ReadonlySet<unknown>,
+  other: ReadonlySet<unknown>,
+): boolean {
+  if (data.size !== other.size) {
+    return false;
+  }
+
+  // TODO: Once we bump our typescript target we can iterate over the map keys and values directly.
+  const dataArr = Array.from(data.values());
+  const otherMap = arrayToMap(Array.from(other.values()));
+  const otherKeys = Array.from(otherMap.keys());
```

Comment:
These structures are overkill. Create an array of the items of the other array like you created for the data (e.g. `Array.from(other.values())` and use splice to remove items from it every time you find a matching element.